cli/src/merge_tools/external.rs

```diff
@@ -61,13 +61,15 @@ pub struct ExternalMergeTool {
     /// If false (default), the `$output` file starts out empty and is accepted
     /// as a full conflict resolution as-is by `jj` after the merge tool is
     /// done with it. If true, the `$output` file starts out with the
-    /// contents of the conflict, with JJ's conflict markers. After the
-    /// merge tool is done, any remaining conflict markers in the
-    /// file parsed and taken to mean that the conflict was only partially
+    /// contents of the conflict, with the configured conflict markers. After
+    /// the merge tool is done, any remaining conflict markers in the
+    /// file are parsed and taken to mean that the conflict was only partially
     /// resolved.
-    // TODO: Instead of a boolean, this could denote the flavor of conflict markers to put in
-    // the file (`jj` or `diff3` for example).
     pub merge_tool_edits_conflict_markers: bool,
+    /// If provided, overrides the normal conflict marker style setting. This is
+    /// useful if a merge tool parses conflict markers, and so it requires a
+    /// specific format.
+    pub conflict_marker_style: Option<ConflictMarkerStyle>,
```

Comment:
Are we going to apply this style to diff generation (i.e. `jj diff` of external tools)? If not, I would rename it to `merge_tool_conflict_marker_style`.

Regarding `merge-tool-edits-conflict-markers`, I'm not sure whether we should merge them into one. If we prefer merging, `merge-tool-edits-conflict-markers: true` can be translated to the default style (maybe with a deprecation warning.)

Reply:
I think it makes sense to apply it for diff generation too, since it seems like a useful option to have. For instance, a diff tool might have syntax highlighting, and it might know how to recognize Git-style conflict markers and highlight them appropriately. I don't feel too strongly about it though, since I think it's less common for diff tools to care about conflict markers than merge tools.